<launch>

  <arg name="model" default="$(find commissioning_robot)/xacro/000_robot_ver4_full_stl.xacro"/>
  <arg name="gui" default="false" />
  <arg name="rvizconfig" default="$(find urdf_tutorial)/rviz/urdf.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

  <!-- <node if="$(arg gui)" name="joint_state_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" /> -->
  <!-- <node unless="$(arg gui)" name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> -->

  <node pkg="joy" name="joy_con" type="joy_node"/>
  <node pkg="commissioning_robot" name="DriveInverseKinematics_node" type="DriveInverseKinematics_node"/>
  <node name="rviz_publisher" pkg="commissioning_robot" type="rviz_publisher" output="screen"/>
  
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>